### ISSUE TITLE
[Bug] stealth Plugin detected on [Gmail&Google] after Login page

### ISSUE THREAD

**Issue by signaldev1** [berstend/puppeteer-extra#668]
Hello, 

since last week this problem has started to show again
![140619288-7953980a-a3d2-49d3-b67f-b3f97bafc102](https://user-images.githubusercontent.com/43824570/177305829-a29a2c97-aa06-4c17-8adf-a8b42e156c1d.png)

It seems like the Gmail has made a new update , 
i use  ` await this.page.setBypassCSP(true)` but not working anymore .

Thank you.

**Comment by opahopa** [comment]
i found that disabling `iframe.contentWindow `, `navigator.plugins` works for me.  enabling any of them results in being detected

**Comment by signaldev1** [comment]
i disable iframe.contentWindow by this code 
```
        const StealthPlugin = require('puppeteer-extra-plugin-stealth')()
        StealthPlugin.enabledEvasions.delete('iframe.contentWindow')
        puppeteer.use(StealthPlugin)

```

and for `navigator.plugins` how to disable it ?

thanks @opahopa

**Comment by kornicameister** [comment]
@signaldev1 there's an evasion called `navigator.plugins` although removing both evasions + removing `setBypassCSP` doesn't help my haedless case.

**Comment by signaldev1** [comment]
@kornicameister  @opahopa  , i confirm that it's working now , Thank you

**Comment by LinkTree3** [comment]
Unfortunately this solution did not work for me :(

I added this
```
const stealthPlugin = StealthPlugin();
stealthPlugin.enabledEvasions.delete('iframe.contentWindow');
stealthPlugin.enabledEvasions.delete('navigator.plugins');
puppeteer.use(StealthPlugin())
```

And I tried it both with `await page.setBypassCSP(true)` and without

Still with the same Gmail error

**Comment by thomask** [comment]
@LinkTree3 It is because in your code you're re-instantiating the Stealthplugin on the `puppeteer.use`  line instead of using the instance on which you've deleted the two evasions.

I can confirm this is currently working:

```javascript
const stealthPlugin = StealthPlugin();
stealthPlugin.enabledEvasions.delete('iframe.contentWindow');
stealthPlugin.enabledEvasions.delete('navigator.plugins');
puppeteer.use(stealthPlugin)
```

**Comment by LinkTree3** [comment]
> @LinkTree3 It is because in your code you're re-instantiating the Stealthplugin on the `puppeteer.use` line instead of using the instance on which you've deleted the two evasions.
> 
> I can confirm this is currently working:
> 
> ```js
> const stealthPlugin = StealthPlugin();
> stealthPlugin.enabledEvasions.delete('iframe.contentWindow');
> stealthPlugin.enabledEvasions.delete('navigator.plugins');
> puppeteer.use(stealthPlugin)
> ```

Yes this is correct! I found it out later but forgot to update here :(

**Comment by signaldev1** [comment]
Once again, the issue has emerged today !
Are there any suggestions or solutions from anyone regarding this problem?

**Comment by rbrzezinski** [comment]
Apparently setting Firefox user agent solves problem:

```js
await page.setUserAgent('Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:121.0) Gecko/20100101 Firefox/121.0');
```

**Comment by mdervisaygan** [comment]
> Once again, the issue has emerged today ! Are there any suggestions or solutions from anyone regarding this problem?

https://github.com/zfcsoftware/puppeteer-real-browser
I created such a library, can you try it?
